### DISCUSSION TITLE
Argument Parser for Inference Mode

### DISCUSSION THREAD

**Discussion by HripsimeS** [Project-MONAI/tutorials#647] [Q&A] [answered]
Dear MONAI team, 

I am trying to see how is possible to modify the inference mode that it would be possible to write on Command Prompt the name of inference script, the name of input image and then receive the output predicted mask. So should look something like this: 

python predict.py -i image.nii.gz -o output.nii.gz

At the moment I can use just "python predict.py" command as the path of the testing images are in my code already. This is the python script of my inference prediction using one image: https://github.com/HripsimeS/Computer-Vision/blob/main/predict.py

I think one of the ways to do it's using Argument Parser.  Can you please let me know how I should modify my code to be able to select one single image from Command Prompt and get its predicted mask. Look forward to hearing from you!

**Comment by Nic-Ma** [comment, marked as answer]
Hi @HripsimeS ,

Thanks for your interest and experiments.
Actually, we are developing a very important feature for the next release: MONAI bundle, which is designed for model package.
Your requirement is exactly one of its CLI command line features. For more details, you can take a look this tutorial:
https://github.com/Project-MONAI/tutorials/blob/master/modules/bundles/get_started.ipynb
Welcome to share your feedback if you are interested to try this experimental feature.

Thanks in advance.

**Reply by HripsimeS** [reply]
@Nic-Ma thanks a lot for the link, I will try and update you later. I will close this ticket now 🥇 👍

**Reply by HripsimeS** [reply]
@Nic-Ma @yiheng-wang-nv 
Hello. In fact I could optimize the script with ArgumentParser, so it can upload one image and then give one output predicted mask. But for some reason I get a RuntimeError applying transform Spacingd, while the same code without ArgumentParser does not show any issue. This is my script:  https://github.com/HripsimeS/Computer-Vision/blob/main/predict.py
On Command Prompt I write the following:    python predict.py -i image.nii.gz -o mask.nii.gz 

When I remove Spicingd transorm, the image is loading but after it gets error on test_inputs = test_data["image"].to(device)

RuntimeError: CUDA out of memory. Tried to allocate 228.00 MiB (GPU 0; 8.00 GiB total capacity; 18.36 MiB already allocated; 6.41 GiB free; 26.00 MiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF

Any ideas why Spacingd transform can be affected by using Argument Parser?  Look forward to hearing from you!

**Reply by Nic-Ma** [reply]
Hi @HripsimeS ,

May I know what's your current error message you want to solve?
I didn't see obvious error in your script.

Thanks.

**Reply by HripsimeS** [reply]
@Nic-Ma I restarted my laptop, then run the short command and it is working now. I did not modify anything, it was probably a Cache which was causing issues for hours. Thanks for your quick reply :)

**Reply by Nic-Ma** [reply]
No problem.

Thanks.
